packages/quantification/src/__tests__/validations.test.ts

```diff
@@ -0,0 +1,334 @@
+import type { Output } from '@nori-dot-com/ggit';
+
+import { SOIL_METRICS_RESULTS_NAME_SUFFIX } from '../constants';
+import { validateParsedModelRunsData } from '../validations';
+
+import { PARSED_SOIL_METRICS_OUTPUT } from './test-fixtures/parsed-model-runs';
+
+// Solving some type uncertainty with model runs being an object or an array
+const mockModelRuns = Array.isArray(
+  PARSED_SOIL_METRICS_OUTPUT.Day.Cropland.ModelRun
+)
+  ? PARSED_SOIL_METRICS_OUTPUT.Day.Cropland.ModelRun
+  : [PARSED_SOIL_METRICS_OUTPUT.Day.Cropland.ModelRun];
+
+describe('validations', () => {
+  describe('validateElevenYearsOfModelRuns', () => {
+    describe('throws an error', () => {
+      it('when the input does not have 2 scenario results', () => {
+        expect(() =>
+          validateParsedModelRunsData({
+            baselineScenarioName: 'Baseline',
+            futureScenarioName: 'Future',
+            modelRuns: [
+              {
+                ...mockModelRuns[0],
+                Scenario: [
+                  {
+                    '@name': `Baseline${SOIL_METRICS_RESULTS_NAME_SUFFIX}`,
+                  },
+                ],
+              },
+            ],
+          })
+        ).toThrow('Expected ModelRun.0 to have 3 results');
+      });
+
+      it('when the input does not have a MapUnit property', () => {
+        expect(() =>
+          validateParsedModelRunsData({
+            baselineScenarioName: 'Baseline',
+            futureScenarioName: 'Future',
+            modelRuns: [
+              {
+                ...mockModelRuns[0],
+                Scenario: [
+                  {
+                    '@name': `Baseline${SOIL_METRICS_RESULTS_NAME_SUFFIX}`,
+                  },
+                  {
+                    '@name': `Current${SOIL_METRICS_RESULTS_NAME_SUFFIX}`,
+                    MapUnit: [],
+                  },
+                  {
+                    '@name': `Future${SOIL_METRICS_RESULTS_NAME_SUFFIX}`,
+                    MapUnit: [],
+                  },
+                ],
+              },
+            ],
+          })
+        ).toThrow(
```

Comment:
Using a custom error definition would make this test less fragile. Some examples in nori-graphql tests I believe